sound/soc/sof/debug.c

```diff
@@ -17,6 +17,142 @@
 #include "sof-priv.h"
 #include "ops.h"
 
+static ssize_t sof_dfsentry_write(struct file *file, const char __user *buffer,
+				  size_t count, loff_t *ppos)
+{
+	struct snd_sof_dfsentry *dfse = file->private_data;
+	struct snd_sof_dev *sdev = dfse->sdev;
+	struct sof_ipc_cmd_hdr hdr;
+	struct sof_ipc_reply reply;
+	size_t size;
+	char *string;
+	int ret;
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_DEBUG_ENABLE_IPC_FLOOD_TEST)
+	ktime_t start, end, test_end;
+	s64 ipc_response_time;
+	s64 avg_response_time = 0;
+	s64 min_response_time = LONG_MAX;
+	s64 max_response_time = 0;
+	int i = 0;
+	int err;
+	unsigned long ipc_count = 0;
+	unsigned long ipc_duration_ms = 0;
+	bool flood_duration_test = false;
```

Comment:
this looks like a mess. can you check that all inits are strictly required and use reverse xmas tree, upstream folks will bark on this one.